include/selene/Registry.h

```diff
@@ -34,17 +35,17 @@ class Registry {
     template <typename Ret, typename... Args>
     void Register(std::function<Ret(Args...)> fun) {
         constexpr int arity = detail::_arity<Ret>::value;
-        auto tmp = std::unique_ptr<BaseFun>(
-            new Fun<arity, Ret, Args...>{_state, _metatables, fun});
-        _funs.push_back(std::move(tmp));
+        _funs.emplace_back(
```

Comment:
Using `emplace_back` instread of `push_back` because the type of the temporary (`unique_ptr<Fun<...>>`) does not match `_funs`'s element type (`unique_ptr<BaseFun>`).